proc/variables.go

```diff
@@ -28,20 +28,47 @@ const (
 
 // Represents a variable.
 type Variable struct {
-	Addr      uintptr
-	Name      string
-	Value     string
-	Type      string
-	dwarfType dwarf.Type
-	thread    *Thread
-
-	Len       int64
-	Cap       int64
+	Addr       uintptr
+	Name       string
+	DwarfType  dwarf.Type
+	RealType   dwarf.Type
+	SimpleType SimpleType
+	thread     *Thread
+
+	Value interface{}
+
+	Len int64
+	Cap int64
+
 	base      uintptr
 	stride    int64
 	fieldType dwarf.Type
+
+	Children []Variable
+
+	loaded     bool
+	Unreadable error
 }
 
+type SimpleType string
+
+const (
```

Comment:
We have access to the dwarf.Entry in `scope.extractVarInfoFromEntry` so why not just parse it there and store it as a field in the `Variable` struct?